src/api/core/tokenManager.spec.ts

```diff
@@ -1,170 +1,286 @@
-      const loginError = "loginError";
-      spyOn(tm as any, "login").and.returnValue(Promise.reject(loginError));
-      try {
-        await tm.init(validOpts());
-        throw new Error("init should have failed!");
-      } catch (error) {
-        expect(error).toBe(loginError);
-      }
+    it("should result promise of itself when authorization succeeded", async () => {
+      const { subject, validOptions } = createSubject();
+      const value = await subject.init(validOptions);
+      expect(value).toEqual(subject);
     });
 
-    it("should result itself at then promise when authorization succeeded", async () => {
-      try {
-        const result = await tm.init(validOpts());
-        expect(result).toBe(tm);
-      } catch (error) {
-        expect(error).not.toBeDefined();
-      }
+    it("should schedule token refreshing when authorization is succeeded", async () => {
+      const { subject, validOptions } = createSubject();
+      jest.spyOn(subject as any, "startRefreshDigest");
+      await subject.init(validOptions);
+      expect((subject as any).startRefreshDigest).toHaveBeenCalledWith("apikey");
     });
 
-    it("should have valid token and refresh token if authorization succeeded", async () => {
-      try {
-        const tokenManager: TokenManager = await tm.init(validOpts());
-        const token: string = await tokenManager.token();
-        expect(token).toBe(defaultToken.access_token);
-      } catch (error) {
-        expect(error).not.toBeDefined();
-      }
+    it("should schedule token refreshing for the authentication by alias", async () => {
+      const { subject, validOptions } = createSubject();
+      const auth = faker.random.word();
+      jest.spyOn(subject as any, "startRefreshDigest");
+      await subject.init({ ...validOptions, auth });
+      expect((subject as any).startRefreshDigest).toHaveBeenCalledWith(auth);
+    });
+  });
+
+  describe("when get a token", () => {
+    it("should return default token if authorization succeeded", async () => {
+      const { subject, validOptions, tokens } = createSubject();
+      await subject.init(validOptions);
+      expect(await subject.token()).toEqual(tokens.access_token);
+    });
+
+    it("should return a token by auth alias if authorized with alias", async () => {
+      const { subject, validOptions, tokens } = createSubject();
+      const randomAlias = faker.random.word();
+      await subject.init({ ...validOptions, auth: randomAlias });
+      expect(await subject.token(randomAlias)).toEqual(tokens.access_token);
+    });
+
+    it("should return a token set by default", async () => {
+      const { subject, validOptions } = createSubject();
+      const randomAlias = faker.random.word();
+      const anotherTokens = {
+        access_token: faker.random.word(),
+        refresh_token: faker.random.word(),
+      };
+      await subject.init(validOptions);
+      (subject as any).storage.setTokens(randomAlias, anotherTokens);
```

Comment:
why do you need to call an external method here? If you still really need to call it I suggest you get the storage from `TokenStorage.getStorageAPI()`

Reply:
fixed